# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/ppoll.patch
# Copyright (C) 2011 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

For udev, too.

  - Rene Rebe <rene@exactcode.de>

--- /dev/null	2010-10-27 14:43:31.488782000 +0200
+++ dietlibc/syscalls.s/__ppoll.S	2011-01-05 15:37:28.000000000 +0100
@@ -0,0 +1,3 @@
+#include "syscalls.h"
+
+syscall(ppoll,__ppoll)
--- /dev/null   2010-10-27 14:43:31.488782000 +0200
+++ dietlibc/lib/ppoll.c     2010-11-13 12:54:09.000000000 +0100
@@ -0,0 +1,11 @@
+#include <sys/types.h>
+#include <fcntl.h>
+#define _GNU_SOURCE
+#include <poll.h>
+
+int __ppoll(struct pollfd *fds, nfds_t nfds, const struct timespec *timeout, const sigset_t *sigmask, long sigsetsize);
+int __libc_ppoll(struct pollfd *fds, nfds_t nfds, const struct timespec *timeout, const sigset_t *sigmask) {
+  return __ppoll(fds, nfds, timeout, sigmask, _NSIG/8);
+}
+int ppoll(struct pollfd *fds, nfds_t nfds, const struct timespec *timeout, const sigset_t *sigmask)
+__attribute__((weak,alias("__libc_ppoll")));
